common/src/update.rs

```diff
@@ -65,6 +74,30 @@ pub struct ArtifactId {
     pub kind: ArtifactKind,
 }
 
+/// A hash-based identifier for an artifact.
+///
+/// Some places, e.g. the installinator, request artifacts by hash rather than
+/// by name and version. This type indicates that.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    Ord,
+    PartialOrd,
+    Deserialize,
+    Serialize,
+    JsonSchema,
+)]
+pub struct ArtifactHashId {
+    /// The kind of artifact this is.
+    pub kind: ArtifactKind,
```

Comment:
If we didn't unpack in wicketd, but instead sent the whole image, and had installinator unpack, would this be a "known" type?. It feels silly to send the whole image to installinator but it also feels silly to not be able to specify a very well known type here.

Reply:
Good point. Well, for now let's stick with this as is and see how it goes :)